node/impl/full/eth.go

```diff
@@ -372,6 +373,65 @@ func (a *EthModule) EthChainId(ctx context.Context) (api.EthUint64, error) {
 	return api.EthUint64(build.Eip155ChainId), nil
 }
 
+func (a *EthModule) EthFeeHistory(ctx context.Context, blkCount uint64, newestBlkNum string) (api.EthFeeHistory, error) {
+	if blkCount > 1024 {
+		return api.EthFeeHistory{}, fmt.Errorf("block count should be smaller than 1024")
+	}
+
+	newestBlkHeight := uint64(a.Chain.GetHeaviestTipSet().Height())
+
+	var blkNum api.EthUint64
+	err := blkNum.UnmarshalJSON([]byte(`"` + newestBlkNum + `"`))
+	if err == nil && uint64(blkNum) < newestBlkHeight {
+		newestBlkHeight = uint64(blkNum)
+	}
```

Comment:
This is be better handled by an `interface{}` type and proper checking of allowed values. See https://github.com/filecoin-project/ref-fvm/issues/1016.